### PR TITLE
Add //os.tree to return a directory/file tree

### LINKED ISSUE #605: Function to get all files in a directory tree
Please **do not post any internal, closed source snippets** on this public issue tracker!

### Purpose

> Please describe the _end goal_ you are trying to achieve that has led you to request this feature.

Std lib function to get all file paths in a directory.

We're starting to work with GitHub as a data source, and want to transform directories, not just files.

`//os.tree` to use `filepath.Walk` and construct some `Value`.
- Array of file path strings?
- Dictionary representing the filesystem?
  - Map dirs to dicts and files to `true`?
  - Represent each entry as a tuple of `os.FileInfo`, with a `dir` property containing a set of children?
- Binary relation with parent -> child?

Whatever structure it returns can be transformed to others, so it should be the structure that is easiest to understand. Probably a FS dict. However a list of file paths would be easier to map into file contents (`//os.tree('dir') => .@item => {.: //os.file(.)}`).

### PR DESCRIPTION
**Pull request by orlade-anz**
Fixes #605 .

Changes proposed in this pull request:
- Add `//os.tree` that returns a set of `FileInfo` tuples representing the file system.
- Tuples also contain `path`

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/std_os_nonwasm.go [hunk 1]
```diff
@@ -58,6 +58,33 @@ func stdOsFile(_ context.Context, v rel.Value) (rel.Value, error) {
...
+
+	fs := []rel.Value{}
+	err := filepath.Walk(d.String(), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		fs = append(fs, rel.NewTuple(
+			rel.NewStringAttr("name", []rune(info.Name())),
+			rel.NewStringAttr("path", []rune(path)),
+			rel.NewBoolAttr("isDir", info.IsDir()),
```
Comment (nofun97) [inline syntax/std_os_nonwasm.go:76]:
```suggestion
			rel.NewBoolAttr("is_dir", info.IsDir()),
```

File: syntax/std_os_nonwasm.go [hunk 2]
```diff
@@ -58,6 +58,33 @@ func stdOsFile(_ context.Context, v rel.Value) (rel.Value, error) {
...
+			rel.NewFloatAttr("size", float64(info.Size())),
+			rel.NewFloatAttr("modTime", float64(info.ModTime().UnixNano())),
```
Comment (nofun97) [inline syntax/std_os_nonwasm.go:78]: for consistency with other stdlib functions
```suggestion
			rel.NewFloatAttr("mod_time", float64(info.ModTime().UnixNano())),
```

**Review by nofun97 — CHANGES_REQUESTED** [review]
also add documentation in here https://github.com/arr-ai/arrai/blob/master/docs/std-os.md

**Comment by orlade-anz** [comment]
Docs already done?

**Pull request merged**